### PR TITLE
[IR-3214] studio: make GizmoTool draggable and fix viewport toolbar z-index 

### PR DESCRIPTION
**Pull request by dtlehrer**
## Summary
[IR-3214]: (1) The GizmoTool should be draggable and placeable anywhere on the viewport, and (2) the viewport toolbar dropdowns should appear above the viewport and its elements

https://github.com/user-attachments/assets/dc438d1c-6c53-4239-95b7-050cdf17b2c5

## References
closes #_insert number here_

## QA Steps

[IR-3214]: https://tsu.atlassian.net/browse/IR-3214?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/ui/src/components/editor/panels/Viewport/tools/GizmoTool.tsx [hunk 1]
```diff
@@ -44,66 +44,125 @@ function Placer() {
...
+        <Tooltip title={t('editor:toolbar.gizmo.translate')} position={'right center'}>
+          <Button
+            variant="transparent"
+            className={twMerge(
+              'border-b border-b-theme-primary p-2 text-[#A3A3A3]',
+              transformMode === TransformMode.translate && 'bg-theme-highlight text-white'
+            )}
+            iconContainerClassName="m-0"
+            startIcon={<TbVector />}
+            title="Translate (key W)"
```
Comment (aditya-mitra) [inline packages/ui/src/components/editor/panels/Viewport/tools/GizmoTool.tsx]: the titles on the buttons are redundant now with the usage of tooltips (and also incorrect). can you remove them on all these 4 buttons?

image

File: packages/ui/src/components/editor/panels/Viewport/tools/GizmoTool.tsx [hunk 2]
```diff
@@ -44,66 +44,125 @@ function Placer() {
   )
 }
 
-export default function GizmoTool() {
+export default function GizmoTool({
```
Comment (aditya-mitra) [inline packages/ui/src/components/editor/panels/Viewport/tools/GizmoTool.tsx:47]: for some reason, i was unable to get the gizmo by clicking on the model. it used to appear just by clicking on the model. 

**dev branch**

https://github.com/user-attachments/assets/673b9183-0692-4bf7-a0db-13d502b1d1c1

**this branch**

https://github.com/user-attachments/assets/09f7870d-6865-43e5-995a-555da04d571a

**Review by aditya-mitra — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/editor/panels/Viewport/tools/GizmoTool.tsx (hunk 2 above)
Comment (dtlehrer) [reply packages/ui/src/components/editor/panels/Viewport/tools/GizmoTool.tsx:47]: good catch, thanks

**Review by aditya-mitra — APPROVED** [review]
awesome

**Comment by DanielBelmes** [comment]
This is really cool

**Pull request merged**